### PR TITLE
ARROW-13494: [C++] Rename BitUtil and LittleEndianArray namespaces

### PR DESCRIPTION
**Pull request by mbrobbel**
Renamed `BitUtil` namespace to `bit_util` and `LittleEndianArray` namespace to `little_endian` to match the suggested style in https://google.github.io/styleguide/cppguide.html#Namespace_Names.

I renamed the `LittleEndianArray` namespace to `little_endian` to handle a symbol conflict with the `little_endian_array` function in `arrow/util/basic_decimal.h`.

### CODE REVIEW

**Comment by pitrou** [comment]
Thanks for submitting this @mbrobbel . Unfortunately it seems the PR has conflicts now, can you rebase?

File: cpp/src/arrow/util/bit_util_test.cc [hunk 1]
```diff
@@ -154,8 +154,8 @@ void ASSERT_BYTES_EQ(const uint8_t* left, const std::vector<uint8_t>& right) {
   ASSERT_EQ(left_array, right);
 }
 
-TEST(BitUtilTests, TestIsMultipleOf64) {
```
Comment (pitrou) [inline cpp/src/arrow/util/bit_util_test.cc:157]: Can you undo the test naming changes here?

**Review by pitrou — CHANGES_REQUESTED** [review]
Thank you for the update. Just one concern here.

File: cpp/src/arrow/util/bit_util_test.cc (hunk 1 above)
Comment (mbrobbel) [reply cpp/src/arrow/util/bit_util_test.cc:157]: Ah, good one. Fixed.

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Since this risks other conflicts I'll try to merge this soon. Waiting for CI.

**Pull request closed without merging**